src/controllers/user/Controller.ts

```diff
@@ -117,6 +119,26 @@ class UserController {
     }
 
 
+    public async remove(req: IRequest, res: Response, next: NextFunction) {
+      const  id  = req.params.id;
+      const remover = req.userData._id;
+      const user = new UserRepository();
+      await user.deleteData(id, remover)
+      .then((result) => {
+          res.send({
+              message: 'Deleted successfully',
+              code: 200
```

Comment:
Don't use both async-await and then() only use one to handle async operation either then() or async-await.